### PR TITLE
write entire request at once

### PR DESCRIPTION
**Pull request by aandyl**
It seems like each write turns into a separate enclave I/O, so, even if the helper service ought to be able to tolerate that, there is a small performance gain from combining them.

### CODE REVIEW

File: aesm-client/src/imp/aesm_protobuf/mod.rs [hunk 1]
```diff
@@ -26,11 +26,14 @@ impl AesmClient {
         #[cfg(not(target_env = "sgx"))]
         let _ = sock.set_read_timeout(req.get_timeout().map(|t| Duration::from_micros(t as _)))?;
 
-        let req_bytes = req
-            .into()
-            .write_to_bytes()
+        let mut req_bytes = vec![0u8; 4];
+        let mut cursor = Cursor::new(&mut req_bytes);
+        cursor.set_position(4);
```
Comment (jethrogb) [inline aesm-client/src/imp/aesm_protobuf/mod.rs]: You don't need a cursor, [`impl io::Write for Vec<u8>` appends to the vector](https://doc.rust-lang.org/1.42.0/std/io/trait.Write.html#impl-Write-9).

File: aesm-client/src/imp/aesm_protobuf/mod.rs [hunk 2]
```diff
@@ -26,11 +26,12 @@ impl AesmClient {
         #[cfg(not(target_env = "sgx"))]
         let _ = sock.set_read_timeout(req.get_timeout().map(|t| Duration::from_micros(t as _)))?;
 
-        let req_bytes = req
-            .into()
-            .write_to_bytes()
+        let mut req_bytes = vec![0u8; 4];
```
Comment (jethrogb) [inline aesm-client/src/imp/aesm_protobuf/mod.rs]: Do you want to use `size_of::<u32>()` in place of `4` everywhere? It's ok if not.

Comment (aandyl) [reply aesm-client/src/imp/aesm_protobuf/mod.rs]: Yeah, I'll update that.

**Comment by jethrogb** [comment]
bors retry

**Pull request merged**